gtrending/cli.py

```diff
@@ -0,0 +1,187 @@
+import argparse
+import json
+
+from typing import Dict, Any
+
+from .fetch import (
+    fetch_repos,
+    fetch_developers,
+    languages_list,
+    spoken_languages_list,
+)
+
+
+def main():
+
+    parser = argparse.ArgumentParser('Gtrending')
+    subparser = parser.add_subparsers(dest='command')
+  
+    # Extract common arguments in parents
+    parent_json = argparse.ArgumentParser(add_help=False)
+    parent_json.add_argument(
+        '--json',
+        '-J',
+        action=argparse.BooleanOptionalAction,
+        default=False,
+    )
+
+    parent_filter = argparse.ArgumentParser(add_help=False)
+    parent_filter.add_argument(
+        '--language',
+        '--lang',
+        default=""
+    )
+    parent_filter.add_argument(
+        '--since',
+        type=str,
+        choices=['daily', 'monthly', 'weekly'],
+        default='daily'
+    )
+
+
+    parser_repo = subparser.add_parser(
```

Comment:
It seems like each of the options (repos/developers/langs/spoken-lang) are considered separate commands, which means their available options can only be shown when `--help` is used on each of the specifically. Would it be possible to show the parent_filter flags at the root-level usage text, since they're shared for repos/developers (or at least only json)?

Reply:
I think showing filters at top level might be confusing as they're only available for two sub-commands. But I agree that `--json` should be visible. I'd tried it but sadly there's no clean way to do this in `argparse`. I've worked around it using two `--json` args, one for top-level command and other for sub-commands.  

```